mathesar_ui/src/pages/record/TableWidget.svelte

```diff
@@ -0,0 +1,99 @@
+<script lang="ts">
+  import type { TableEntry } from '@mathesar/api/tables';
+  import type { Column } from '@mathesar/api/tables/columns';
+  import { Icon } from '@mathesar/component-library';
+  import Identifier from '@mathesar/components/Identifier.svelte';
+  import { iconMultipleRecords } from '@mathesar/icons';
+  import {
+    setTabularDataStoreInContext,
+    TabularData,
+  } from '@mathesar/stores/table-data/tabularData';
+  import TableView from '@mathesar/systems/table-view/TableView.svelte';
+  import { currentDbAbstractTypes } from '@mathesar/stores/abstract-types';
+  import { Filtering, Meta } from '@mathesar/stores/table-data';
+  import Pagination from '@mathesar/utils/Pagination';
+  import MiniActionsPane from '@mathesar/systems/table-view/actions-pane/MiniActionsPane.svelte';
+
+  const tabularDataStore = setTabularDataStoreInContext(
+    // Sacrifice type safety here since the value is initialized reactively
+    // below.
+    undefined as unknown as TabularData,
+  );
+
+  export let recordId: number;
+  export let table: Pick<TableEntry, 'id' | 'name'>;
+  export let fkColumn: Pick<Column, 'id' | 'name'>;
+
+  $: meta = new Meta({
+    pagination: new Pagination({ size: 10 }),
```

Comment:
I propose we hide the size selector in pagination for table widgets and limit the rows to be 10 always. Rendering a large number of rows in the record page may not be ideal since we do not use a virtual list here and using a virtual list would not be possible without having a scrollbar within each table widget.

Reply:
That's good. I like that idea. I've updated #1580 to include this